pkg/api/handlers/mcpgateway/oauth/authorize.go

```diff
@@ -237,34 +238,46 @@ func (h *handler) callback(req api.Context) error {
 	}
 
 	// Check whether the MCP server needs authentication.
-	mcpServer, mcpServerConfig, err := handlers.ServerFromMCPServerInstance(req, h.gptClient)
+	var (
+		mcpID           = req.PathValue("mcp_id")
+		mcpServer       v1.MCPServer
+		mcpServerConfig mcp.ServerConfig
+		err             error
+	)
+
+	if strings.HasPrefix(mcpID, system.MCPServerInstancePrefix) {
+		mcpServer, mcpServerConfig, err = handlers.ServerFromMCPServerInstance(req, h.gptClient, mcpID)
+	} else {
+		mcpServer, mcpServerConfig, err = handlers.ServerForActionWithID(req, h.gptClient, mcpID)
+	}
```

Comment:
nit: this logic is duplicated in a couple places. Does that make it worth adding another function for this?